### ISSUE TITLE
[Xfce] Thunar (list view + status bar enabled) Pressing the 'End' key doesn't go to the actual end if the last file has ー or ／ or 〜 in its file name AND there are enough files that the file in question is far off-screen

### ISSUE THREAD

**Issue by NintendoManiac64** [linuxmint/mint21-beta#95]
(tested in a live session of Xfce as well as a fully-updated local installation as of 2022-07-15)

**Describe the bug**
In Thunar set to "list view" with the status bar enabled, pressing the 'End' key doesn't go to the actual end if the last file has ー or ／ or 〜 in its file name and there are enough files that the file in question is far off-screen (e.g. you would normally have to scroll an additional full page's length to reach the file)

Note that the symbols ー and ／ and 〜 are special unicode character and not their normal ANSI equivalents of - and / and ~

**To Reproduce**
1. boot into Mint 21 beta Xfce (whether a local installation or a live session)
2. Open Thunar
3. Make sure Thunar is set to "List View" and is using it by default
4. navigate into a folder with a _lot_ of files, such as `/etc/`
5. (if using `/etc/` then open the folder as root)
6. create a new blank file with one of the following file names _or_ move/copy existing files with one of the following names into the current folder:
- `zzz ー test`
- `zzz ／ test`
- `zzz 〜 test`
7. scroll to the top of the current folder
8. go back or "Up" to the parent folder (in this example that'd be / aka the root of the drive)
9. once again go into `/etc`
10. press the "End" key on your keyboard
11. ...note how Thunar does not actually scroll all the way to the bottom but rather to a bit above the bottom

**Expected behavior**
Thunar should scroll to the very bottom when the "End" key is pressed regardless of what the file name is of the last file.

**Comment by mtwebster** [comment]
I can't reproduce this - are you sure it's not a horizontal overlay scrollbar getting in the way?

Regardless, this would be something for the Thunar maintainers to address.

**Comment by NintendoManiac64** [comment]
> I can't reproduce this - are you sure it's not a horizontal overlay scrollbar getting in the way?

Absolutely as the issue has it positioned noticeably off-screen (like by two or three items), and I have been able to very easily replicate it over and over.

That is still very peculiar that you cannot replicate it - I wonder if there's a step somewhere I missed, or the bug factors in things like screen resolution...

OH CRAP. I just realized that I forgot to mention to set Thunar to use "List View" in the actual steps! (though I mentioned it in the initial "Describe the bug")  I don't suppose that's why you're unable to replicate it? Basically, after step 2, make sure Thunar is set to always using "List View".

I've now edited the steps to include an additional step mentioning "List View".
